0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md

```diff
@@ -1,73 +1,56 @@
 
 ## Introduction
 
 ### Motivation
 
-STREAM does not specify how payment details, such as the ILP address or shared secret, should be exchanged between the sender and receiver. SPSP is a minimal protocol that uses HTTPS for communicating these details.
+STREAM does not specify how payment details, such as the ILP address or shared secret, should be exchanged between the counterparties. SPSP is a minimal protocol that uses HTTPS for communicating these details.
 
 ### Scope
 
-SPSP provides for exchanging basic receiver details needed by a sender to set up a STREAM payment. It is intended for use by end-user applications.
+SPSP provides for exchanging basic server details needed by a client to set up a STREAM connection. It is intended for use by end-user applications.
 
 ### Interfaces
 
-SPSP may be used by end-user applications, such as a digital wallet with a user interface for the sender to initiate payments. SPSP clients and receivers use ILP modules to send and receive Interledger payments. SPSP [payment-pointers](../0026-payment-pointers/0026-payment-pointers.md) can be used as a persistent identifier on Interledger. SPSP payment-pointers can also be used as a unique identifier for an invoice to be paid.
+SPSP may be used by end-user applications, such as a digital wallet with a user interface to initiate payments. SPSP clients and servers use ILP modules to send and receive Interledger payments. SPSP [payment-pointers](../0026-payment-pointers/0026-payment-pointers.md) can be used as a persistent identifier on Interledger. SPSP payment-pointers can also be used as a unique identifier for an invoice to be paid or for a pull payment agreement. 
 
 SPSP messages MUST be exchanged over HTTPS.
 
 ### Operation
 
-Any SPSP receiver will run an SPSP server and expose an HTTPS endpoint called the SPSP Endpoint. The sender can query this endpoint to get information about the type of payment that can be made to this receiver. The sender can set up and send multiple ILP payments using the details provided by the receiver.
+Any SPSP server will expose an HTTPS endpoint called the SPSP Endpoint. The client can query this endpoint to get information about the server's connection details, namely ILP address and a shared secret. The client can use this information to establish a STREAM connection.
 
 ### Definitions
 
-* **SPSP Client** - The sender application that uses SPSP to interact with the SPSP Server
-* **SPSP Server** - The server used on the receiver's side to handle SPSP requests
-* **SPSP Endpoint** - The specific HTTPS endpoint on the SPSP server used for setting up a payment
-* **STREAM Module** - Software included in the SPSP Client and Server that implements the [STREAM](../0029-stream/0029-stream.md) protocol.
-
-## Overview
+* **SPSP Client** - The entity initiating a STREAM connection to the SPSP server
```

Comment:
I was getting ahead of myself here. You are right, it should be 

> The entity initiating an SPSP connection to the SPSP server

I have always thought of seen the word _entity_ as a collective term for the person and the software because the person cannot do anything without the software and the software will probably not run without the person telling it to do so. 

I'm not experienced with writing specs so I'm happy to change it to 

> **SPSP Client** - The **software** initiating a SPSP connection to the SPSP server.

or whatever seems to be more fitting.

Reply:
> I have always thought of seen the word _entity_ as a collective term for the person and the software

I agree. My point here is that we talk about two protocols in this spec; SPSP and STREAM. They SPSP client (HTTP client software) and STREAM client software may be run by the same entity so it seems confusing to call that entity the _SPSP Client_.

We do need a way to refer to the _entity_ thought too.

Reply:
Ok, I see your point. I'll try to make it less ambiguous.